include/mbedtls/ssl.h

```diff
@@ -4723,6 +4748,11 @@ int mbedtls_ssl_get_session(const mbedtls_ssl_context *ssl,
  * \return         #MBEDTLS_ERR_SSL_HELLO_VERIFY_REQUIRED if DTLS is in use
  *                 and the client did not demonstrate reachability yet - in
  *                 this case you must stop using the context (see below).
+ * \return         #MBEDTLS_ERR_SSL_RECEIVED_EARLY_DATA if early data, as
```

Comment:
For the record (as it was discussed yesterday but only but not in writing): expanding an existing function with a new return value that requires special handling is a compatibility smell and should always give us pause.

In this instance however it's fine as the new behaviour is opt-in: the application can only get this returned if it called `mbedtls_ssl_tls13_conf_early_data()` earlier. That means existing applications that used to ignore the possibility of early data will keep working as they should.

However, I think we should mention in the documentation of `mbedtls_ssl_tls13_conf_early_data()` that an application that enables early data needs to be prepared to handle `MBEDTLS_ERR_SSL_RECEIVED_EARLY_DATA` being returned by <list of functions that can now return it>. This kind of action at a distance is not obvious so it should be mentioned in the place that causes it. (If I was an existing user updating my code to support ealy data, I'd probably read the documentation of the new functions, but I most probably wouldn't go and re-read the documentation of every existing function just to check if it gained new behaviour related to early data).